### PR TITLE
xosan3

### PR DESCRIPTION
**Pull request by nraynaud**
this is where the next iteration of Xosan is going, do not merge yet.

### CODE REVIEW

File: src/api/pif.js [hunk 1]
```diff
@@ -60,7 +60,7 @@ connect.resolve = {
 // ===================================================================
 // Reconfigure IP
 
-export async function reconfigureIp ({ pif, mode = 'DHCP', ip, netmask, gateway, dns }) {
+export async function reconfigureIp ({ pif, mode = 'DHCP', ip = '', netmask = '', gateway = '', dns = '' }) {
```
Comment (olivierlambert) [inline src/api/pif.js]: I think this is related to the issue https://github.com/vatesfr/xo-web/issues/2346

File: src/api/xosan.js [hunk 2]
```diff
@@ -450,85 +545,118 @@ async function mountNewDisk (localEndpoint, hostname, newDeviceFiledeviceFile) {
...
+    await xapi.xo.setData(xosansr, 'xosan_config', data)
+    CURRENT_POOL_OPERATIONS[poolId] = {...OPERATION_OBJECT, state: 3}
+    await umountDisk(localEndpoint, previousBrickRoot)
+    const previousVBD = previousVM.$VBDs.find(vbd => vbd.device === previousBrickDevice)
+    await xapi.disconnectVbd(previousVBD)
+    await xapi.deleteVdi(previousVBD.VDI)
+    CURRENT_POOL_OPERATIONS[poolId] = {...OPERATION_OBJECT, state: 4}
+    await xapi.call('SR.scan', xapi.getObject(xosansr).$ref)
+  } finally {
+    delete CURRENT_POOL_OPERATIONS[poolId]
```
Comment (julien-f) [inline src/api/xosan.js]: FYI, you can use `defer` for this, similarly to `defer.onFailure`.

File: src/api/xosan.js [hunk 3]
```diff
@@ -721,60 +860,61 @@ export const removeBricks = defer.onFailure(async function ($onFailure, { xosans
...
-  { layout: 'replica', redundancy: 2, capacity: 7 }]
+  {layout: 'disperse', redundancy: 6, capacity: 8},
+  {layout: 'replica', redundancy: 2, capacity: 7}]
 POSSIBLE_CONFIGURATIONS[15] = [
-  { layout: 'disperse', redundancy: 7, capacity: 8 },
-  { layout: 'replica', redundancy: 3, capacity: 5 }]
-POSSIBLE_CONFIGURATIONS[16] = [{ layout: 'replica', redundancy: 2, capacity: 8 }]
+  {layout: 'disperse', redundancy: 7, capacity: 8},
+  {layout: 'replica', redundancy: 3, capacity: 5}]
+POSSIBLE_CONFIGURATIONS[16] = [{layout: 'replica', redundancy: 2, capacity: 8}]
```
Comment (julien-f) [inline src/api/xosan.js]: Why all this whitespace changes?

File: src/api/xosan.js (hunk 2 above)
Comment (nraynaud) [reply src/api/xosan.js]: I'm always on the fence with defer and onFailure, they avoid indenting to far right, but they are not visible.

File: src/api/xosan.js (hunk 3 above)
Comment (nraynaud) [reply src/api/xosan.js]: IDE/standard issue.

**Pull request merged**